internal/datanode/services.go

```diff
@@ -227,22 +232,25 @@ func (node *DataNode) CompactionV2(ctx context.Context, req *datapb.CompactionPl
 		task = compaction.NewLevelZeroCompactionTask(
 			taskCtx,
 			binlogIO,
-			node.allocator,
 			node.chunkManager,
 			req,
 		)
 	case datapb.CompactionType_MixCompaction:
+		if req.GetPreAllocatedSegments() == nil || req.GetPreAllocatedSegments().GetBegin() == 0 {
```

Comment:
naming seems to be really weird. 

PreAllocatedSegments -> PreAllocatedIDs or something?

Reply:
Again I didn't see anywhere we are using it, why bother check it now?